wally-registry-backend/src/auth.rs

```diff
@@ -73,7 +121,40 @@ impl<'r> FromRequest<'r> for ReadAccess {
 }
 
 pub struct WriteAccess {
-    _dummy: i32,
+    github: Option<GithubInfo>,
+}
+
+impl WriteAccess {
+    pub fn github(&self) -> &Option<GithubInfo> {
+        &self.github
+    }
+
+    pub fn can_write_package(
```

Comment:
I think this is a little more than `can_write_package` because it also mutates the index if the user can write. In the future, we might have an API endpoint for the frontend where we might want to just _check_ if the user can write.

Additionally, `can_do_x` would usually return a `bool`! This is more like... "let me write to this package"

What's a better name for this method?

Reply:
This is a good point! I've kept the name changed this to return a bool for "if the user should be allowed to write a package". Ie they can have permission but not be an owner. We therefore then later check to see if they are an owner and if they aren't we add them as we know they have authorization to write. This seems to make far more semantic sense!